specification/web/resource-manager/Microsoft.Web/2016-03-01/Diagnostics.json

```diff
@@ -0,0 +1,1532 @@
+          "description": "Get Site Analyses",
+          "operationId": "Diagnostics_ListSiteAnalyses",
+          "parameters": [
+            {
+              "$ref": "#/parameters/resourceGroupNameParameter"
+            },
+            {
+              "name": "siteName",
+              "in": "path",
+              "description": "Site Name",
+              "required": true,
+              "type": "string"
+            },
+            {
+              "name": "diagnosticCategory",
+              "in": "path",
+              "description": "Diagnostic Category",
+              "required": true,
+              "type": "string"
+            },
+            {
+              "$ref": "#/parameters/subscriptionIdParameter"
+            },
+            {
+              "$ref": "#/parameters/apiVersionParameter"
+            }
+          ],
+          "responses": {
+            "200": {
+              "description": "OK",
+              "schema": {
+                "$ref": "#/definitions/DiagnosticAnalysisCollection"
+              }
+            }
+          },
+          "x-ms-pageable": {
+            "nextLinkName": "nextLink"
+          }
+        }
+      },
+      "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{siteName}/diagnostics/{diagnosticCategory}/analyses/{analysisName}": {
+        "get": {
+          "tags": [
+            "Diagnostics"
+          ],
+          "summary": "Get Site Analysis",
+          "description": "Get Site Analysis",
+          "operationId": "Diagnostics_GetSiteAnalysis",
+          "parameters": [
+            {
+              "$ref": "#/parameters/resourceGroupNameParameter"
+            },
+            {
+              "name": "siteName",
+              "in": "path",
+              "description": "Site Name",
+              "required": true,
+              "type": "string"
+            },
+            {
+              "name": "diagnosticCategory",
+              "in": "path",
+              "description": "Diagnostic Category",
+              "required": true,
+              "type": "string"
+            },
+            {
+              "name": "analysisName",
+              "in": "path",
+              "description": "Analysis Name",
+              "required": true,
+              "type": "string"
+            },
+            {
+              "$ref": "#/parameters/subscriptionIdParameter"
+            },
+            {
+              "$ref": "#/parameters/apiVersionParameter"
+            }
+          ],
+          "responses": {
+            "200": {
+              "description": "OK",
+              "schema": {
+                "$ref": "#/definitions/DiagnosticAnalysis"
+              }
+            }
+          }
+        }
+      },
+      "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{siteName}/diagnostics/{diagnosticCategory}/analyses/{analysisName}/execute": {
```

Comment:
What will this POST call do? Can PUT on /analysis/{analyisName} not create or execute it?

Reply:
You cannot create it. These are basically predefined analyses that you can run on your web app. I used the post call to execute the analysis. Would it be more correct to use the PUT verb for this purpose?